src/Evolution/Executables/ScalarWave/EvolveScalarWave.hpp

```diff
@@ -71,19 +75,27 @@ struct EvolutionMetavars {
   using boundary_condition_tag = analytic_solution_tag;
   using normal_dot_numerical_flux =
       OptionTags::NumericalFluxParams<ScalarWave::UpwindFlux<Dim>>;
+
+  // public for use by the Charm++ registration code
+  using events = tmpl::list<dg::Events::Registrars::Observe<
+      Dim, db::get_variables_tags_list<typename system::variables_tag>,
+      db::get_variables_tags_list<typename system::variables_tag>>>;
+  using triggers = Triggers::time_triggers;
+
   // A tmpl::list of tags to be added to the ConstGlobalCache by the
   // metavariables
   using const_global_cache_tag_list =
       tmpl::list<analytic_solution_tag,
                  OptionTags::TypedTimeStepper<tmpl::conditional_t<
-                     local_time_stepping, LtsTimeStepper, TimeStepper>>>;
+                     local_time_stepping, LtsTimeStepper, TimeStepper>>,
+                 OptionTags::EventsAndTriggers<events, triggers>>;
   using domain_creator_tag = OptionTags::DomainCreator<Dim, Frame::Inertial>;
 
   struct ObservationType {};
   using element_observation_type = ObservationType;
 
   using observed_reduction_data_tags = observers::collect_reduction_data_tags<
-      tmpl::list<ScalarWave::Actions::Observe>>;
+      typename Event<events>::creatable_classes>;
```

Comment:
You don‘t need the `typename` in the other executables, do you need it here?

Reply:
Yes.  In the others `events` is a definite type so it is immediately known that `Event<events>::creatable_classes` is a type, but here `events` depends on `Dim` so `Event<events>::creatable_classes` cannot be looked up when the `EvolutionMetavars` class template is being parsed.

Reply:
Ah, thanks!